Scenarios/run_Scenario.bat

```diff
@@ -1,74 +1,211 @@
+FOR /L %%i in (1,1,%Filesx%) DO ECHO [%%i]: "!list[%%i]!"
+:again_esmini
+ECHO ------------------------------------
+ECHO Select a scenario by number.
 ECHO Abort with 0.
-ECHO ====================================
-set id=0
-set /p id="Enter ID: "
+SET id=0
+SET /p id="Enter ID: "
 
-if %id% GEQ 1 (
-    if %id% LEQ !x! (
+IF %id% GEQ 1 (
+    IF %id% LEQ !x! (
         "%ESMINI%\EgoSimulator.exe" --window 100 200 1024 576  --osc !list[%id%]! --trails off
-        GOTO again
+        GOTO again_esmini
+    )
+)
+
+GOTO finish
+
+:OpenPASS
+
+SET num_supported_scenario_version=7
+SET list_supported_scenario_version[1]=4.1_1
+SET list_supported_scenario_version[2]=4.2_1
+SET list_supported_scenario_version[3]=4.2_2
+SET list_supported_scenario_version[4]=4.2_4
+SET list_supported_scenario_version[5]=4.5_1
+SET list_supported_scenario_version[6]=4.5_2
+SET list_supported_scenario_version[7]=4.6_1
+
+ECHO ####################################
+ECHO #    OPENPASS: SYSTEM SETTINGS     #
+ECHO ####################################
+
+SET dir_path=%~dp0
+
+IF DEFINED OPENPASS (
+    ECHO OPENPASS path found.
+    ECHO OPENPASS path: %OPENPASS%
+) ELSE (
+    ECHO OPENPASS path not found. 
+    ECHO Please create OPENPASS environment variable, which directs to the installation directory of openPASS and restart shell.
+    PAUSE 
+    GOTO finish
+)
+
+:: Missing: Version Check of OpenPASS
+
+ECHO Settings successful.
+
+SET x=0
+FOR /F "delims=" %%a in ('dir /s /b %dir_path%*.xosc') DO (
+    SET /A "x+=1"
+    SET list_scenario_path_and_filename[!x!]=%%a
+    SET list_scenario_path[!x!]=%%~dpa
+    SET list_scenario_filename[!x!]=%%~nxa
+)
+SET num_scenarios=%x%
+
+SET x=0
+FOR /L %%i in (1,1,%num_scenarios%) DO (
+    FOR /L %%j in (1,1,%num_supported_scenario_version%) DO (
+        IF "!list_scenario_filename[%%i]:~14,5!"=="!list_supported_scenario_version[%%j]!" (
+            SET /A "x+=1"
+            SET list_supported_scenario_path_and_filename[!x!]=!list_scenario_path_and_filename[%%i]!
+        )
     )
 )
 
-:finish
+ECHO ####################################
+ECHO #      OPENPASS: SIMULATION        #
+ECHO ####################################
+
+ECHO Supported OpenSCENARIO (XOSC) files:
+FOR /L %%i in (1,1,%num_supported_scenario_version%) DO (
+    ECHO [%%i]: !list_supported_scenario_path_and_filename[%%i]!
+)
+
+:again_openpass
+ECHO ------------------------------------
+ECHO Select a scenario by number.
+ECHO Abort with 0.
+SET id=0
+SET /p id="Enter ID: "
+
+if %id% GEQ 1 (
+    if %id% LEQ !num_supported_scenario_version! (
+        ECHO Running...
+        IF EXIST %OPENPASS%\configs\NUL (
+            ECHO Config directory does exist. Empty config directory.
+            cd /d "%OPENPASS%\configs"
+            FOR /F "delims=" %%i in ('dir /b') DO (
+                rmdir "%%i" /s/q 2>NUL || del "%%i" /s/q >NUL 
+            )
+        ) ELSE (
+            ECHO Config directory does not exist. Create config directory.
```

Comment:
Creating config directory.

Reply:
Resolved in a2aaf59093454ee9a5886ba5cfb856c760cdf850